Let date be a date, not just datetime; also default _at() methods to now()

jvavrek opened the pull request:
Closes #226

cosama commented:
Looks good to me. Might make sense to add a test for the `datetime.date` thing.

Also there is this test: https://github.com/lbl-anp/becquerel/blob/update-date/tests/isotope_qty_test.py#L249. We could add no argument to it.

Looking at that I realized that there is a `*_now` method as well. Maybe can deprecate that and simplify the interface by defaulting to now with the `*_at` methods. @markbandstra Any thoughts?

jvavrek commented:
@cosama this should be good to go:
 - added a test for the `datetime` parse
 - added no argument to `test_isotopequantity_activity_now()`
 - deprecated the `_now()` methods

cosama commented on tests/isotope_qty_test.py:
```diff
@@ -244,10 +256,19 @@ def test_isotopequantity_time_when_error(stable_isotope):
+    with pytest.warns(DeprecationWarning):
+        assert np.isclose(iq.g_now(), iq.g_at(datetime.datetime.now()))
+
+    assert np.isclose(iq.bq_at(), iq.bq_at(datetime.datetime.now()))
```
These tests might break on very slow machine. They only work right now because `rtol=1e08` by default for isclose. I think it would be better to define an absolute tolerance as a function of of how much decay we would expect in lets say a second.

```
assert np.isclose(iq.bq_at() - iq.bq_at(datetime.datetime.now()), iq.bq_during(iq.ref_time, iq.ref_time + datetime.timedelta(0,1)))
```
Not very pretty but probably safer. Feel free to change this if you see a better way of implementing it.

jvavrek replied:
Ah yeah, I figured this could be the case. I'll think about this this afternoon.

jvavrek replied:
The issue with using 1 s is that 1 s can mean very different activities at `now` vs at the `ref_time`

jvavrek replied:
We could possibly do a speed test and `@pytest.mark.skipif` if the two `datetime.datetime.now()` are too different. Or we could just loosen the `rtol` in `np.isclose`

cosama replied:
`rtol` is very generous for large numbers, not sure if it is needed. Maybe it is fine as it is.

markbandstra replied:
How about this sequence:
```
x0 = iq.bq_at()
x1 = iq.bq_at(datetime.datetime.now())
x2 = iq.bq_at()
```
and test that `x2 <= x1 <= x0`.

jvavrek replied:
done

cosama commented:
just realize there is only `ref_atoms` but no `ref_bq` or other units. Probably worth adding that? Here or new one?

markbandstra reviewed (approved):
Thanks!

jvavrek commented:
> just realize there is only `ref_atoms` but no `ref_bq` or other units. Probably worth adding that? Here or new one? Maybe a `ref_quantity` as well?

The `ref_atoms` docstring says:
```
Access the reference atoms directly (for backwards compatibility)
```
so I assume that `self._ref_quantities` is the preferred way.
